<launch>
  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />

    <!-- load the controllers -->
  <rosparam file="$(find controller_g)/config/config.yaml" command="load"/>
  <node name="controller_spawner" pkg ="controller_manager" type="spawner" ns="$(arg UAV_NAME)" args="gripper_mecha_mount_stretcher_joint_controller gripper_mecha_right_joint_controller gripper_mecha_left_joint_controller joint_state_controller"/>
  
  <!-- converts joint states to TF transforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
      <remap from="joint_state" to="/$(arg UAV_NAME)/joint_states" />
      <remap from="robot_description" to="/$(arg UAV_NAME)/robot_description" />
  </node>

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_ControllerG" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">
  
    <node pkg="nodelet" type="nodelet" name="ControllerG" args="$(arg nodelet) controllerg/ControllerG $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">
      <param name="uav_name" value="$(arg UAV_NAME)" />
    </node>

  </group>
</launch>